windows/src/engine/kmtip/kmkey.cpp

```diff
@@ -198,6 +209,11 @@ STDAPI CKeymanEditSession::DoEditSession(TfEditCookie ec)
     _hr = E_FAIL; // TODO: use S_FALSE -> this tells _KeymanProcessKeystroke to pass keystroke on
   }
 
+  if (!Keyman32Interface::TIPProcessKeyEx(_wParam, _lParam, ExtKeymanProcessOutput, ExtKeymanGetContextIsSelected, _fUpdate, _fPreserved)) {
+    SendDebugMessage(L"TIPProcessKeyEx did not handle the keystroke");
+    _hr = E_FAIL; // TODO: use S_FALSE -> this tells _KeymanProcessKeystroke to pass keystroke on
+  }
```

Comment:
This code is puzzling to me -- why are you calling both TIPProcessKey and TIPProcessKeyEx? Shouldn't be either one or the other?

Reply:
Oh This catch I missed this one. I don't think I can tell the interface available here so I will just have the use the new one.

Reply:
With our new code we should always be calling the new interface. The old interface is only there for the in-memory upgrade scenario to prevent mass crashes, right? We never want to ever call it again from new code